src/rules/no-one-iteration-loop.ts

```diff
@@ -0,0 +1,69 @@
+import { Rule } from "eslint";
+import { Node } from "estree";
+import { isContinueStatement, getParent, isWhileStatement, isForStatement } from "../utils/nodes";
+
+const rule: Rule.RuleModule = {
+  create(context: Rule.RuleContext) {
+    const loopingNodes: Set<Node> = new Set();
+    const loops: Set<Node> = new Set();
+    const loopsAndTheirSegments: { loop: Node; segments: Rule.CodePathSegment[] }[] = [];
+
+    let currentCodePaths: Rule.CodePath[] = [];
+
+    return {
+      ForStatement(node: Node) {
+        loops.add(node);
+      },
+      WhileStatement(node: Node) {
+        loops.add(node);
+      },
+      DoWhileStatement(node: Node) {
+        loops.add(node);
+      },
+
+      onCodePathStart(codePath: Rule.CodePath) {
+        currentCodePaths.push(codePath);
+      },
+
+      onCodePathEnd() {
+        currentCodePaths.pop();
+      },
+
+      "*"() {
```

Comment:
you can use two selectors `WhileStatement > *` and `ForStatement > *` to:
- not getting ancestors of each node
- not check `if (isWhileStatement(parent) || isForStatement(parent))`